src/material-experimental/mdc-table/table-data-source.ts

```diff
@@ -0,0 +1,340 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// NOTE: This file is a direct copy of src/material/table/table-data-source.ts, but it imports
+// `MatPaginator` from `@angular/material-experimental/mdc-paginator` rather than
+// `@angular/material/paginator`.
+// TODO(mmalerba): Refactor this into a common base class with the non-MDC version.
+
+import {_isNumberValue} from '@angular/cdk/coercion';
+import {DataSource} from '@angular/cdk/table';
+import {
+  BehaviorSubject,
+  combineLatest,
+  merge,
+  Observable,
+  of as observableOf,
+  Subscription,
+  Subject,
+} from 'rxjs';
+import {MatPaginator, PageEvent} from '@angular/material-experimental/mdc-paginator';
+import {MatSort, Sort} from '@angular/material/sort';
+import {map} from 'rxjs/operators';
+
+/**
+ * Corresponds to `Number.MAX_SAFE_INTEGER`. Moved out into a variable here due to
+ * flaky browser support and the value not being defined in Closure's typings.
+ */
+const MAX_SAFE_INTEGER = 9007199254740991;
+
+/**
+ * Data source that accepts a client-side data array and includes native support of filtering,
+ * sorting (using MatSort), and pagination (using MatPaginator).
+ *
+ * Allows for sort customization by overriding sortingDataAccessor, which defines how data
+ * properties are accessed. Also allows for filter customization by overriding filterTermAccessor,
+ * which defines how row data is converted to a string for filter matching.
+ *
+ * **Note:** This class is meant to be a simple data source to help you get started. As such
+ * it isn't equipped to handle some more advanced cases like robust i18n support or server-side
+ * interactions. If your app needs to support more advanced use cases, consider implementing your
+ * own `DataSource`.
+ */
+export class MatTableDataSource<T> extends DataSource<T> {
```

Comment:
Shouldn't we re-export the Material one instead of copying it over?

Reply:
I need to change what version of `MatPaginator` is imported. I will make a generic base class in a followup, but doing this for expediency wrt syncing into g3